packages/components/src/modal/Modal.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Modal from './Modal';
+
+function makeModal({
+  className,
+  children,
+  keyboard,
+  isOpen,
+  centered,
+  onOpened,
+  toggle,
+  'data-testid': dataTestId,
+}: {
+  className?: string;
+  children?;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  onOpened?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}) {
+  return (
+    <Modal
+      className={className}
+      keyboard={keyboard}
+      isOpen={isOpen}
+      centered={centered}
+      onOpened={onOpened}
+      toggle={toggle}
+    >
+      {children}
+    </Modal>
+  );
+}
+
+it('mounts', () => {
+  render(makeModal({}));
+});
+
+it('does not render when isOpen is false', () => {
+  render(makeModal({ isOpen: false }));
+  expect(screen.queryByRole('dialog')).toBeNull();
+});
+
+it('renders when isOpen is true', async () => {
+  render(makeModal({ isOpen: true }));
+  const dialogs = screen.getAllByRole('dialog');
+  expect(dialogs.length).not.toBe(0);
+});
+
+it('renders when isOpen changes', async () => {
+  const { rerender } = render(makeModal({ isOpen: false }));
+
+  expect(screen.queryByRole('dialog')).toBeNull();
+  rerender(makeModal({ isOpen: true }));
+
+  expect(screen.getAllByRole('dialog').length).not.toBe(0);
+  rerender(makeModal({ isOpen: false }));
+
+  await waitFor(() => expect(screen.queryByRole('dialog')).toBeNull());
```

Comment:
Put a comment here about how it's waiting for the animation to complete.

Or - another way you could write this is by using fake timers. `waitFor` works but it polls every 50ms, which if we had a lot of these kinds of tests it could cause them to run for a long time. We don't want to _actually_ wait the 150ms it takes for the animation to complete to know if the tests pass or not.

Instead, use fake timers, and then run all pending timers prior to checking. That will ensure the tests run as quickly as possible:

```
it('renders when isOpen changes', () => {
  jest.useFakeTimers();

  const { rerender } = render(makeModal({ isOpen: false }));

  expect(screen.queryByRole('dialog')).toBeNull();
  rerender(makeModal({ isOpen: true }));

  expect(screen.getAllByRole('dialog').length).not.toBe(0);
  rerender(makeModal({ isOpen: false }));

  // It will fade out, so it will still be visible until the animation is complete
  expect(screen.queryAllByRole('dialog').length).not.toBe(0);
  jest.runAllTimers();
  expect(screen.queryAllByRole('dialog').length).toBe(0);

  jest.useRealTimers();
});
```